service/raft/group0_state_machine.cc

```diff
@@ -124,16 +124,47 @@ bool should_flush_system_topology_after_applying(const mutation& mut, const data
     return false;
 }
 
-future<> write_mutations_to_database(storage_proxy& proxy, gms::inet_address from, utils::chunked_vector<canonical_mutation> cms) {
+static void collect_client_routes_update(const mutation& mut, const storage_proxy& proxy, std::vector<client_routes_service::client_route_key>& client_routes_update) {
+    if (!proxy.data_dictionary().has_schema(db::system_keyspace::NAME, db::system_keyspace::CLIENT_ROUTES)) {
+        return;
+    }
+
+    auto s_client_routes = db::system_keyspace::client_routes();
+    if (mut.column_family_id() != s_client_routes->id()) {
+        return;
+    }
+
+    const auto pk_components = mut.decorated_key()._key.explode(*s_client_routes);
+    if (pk_components.empty()) {
```

Comment:
maybe some comment what kind of case is this?

Reply:
> for things which we know can't happen we're typically using SCYLLA_ASSERT

This is a dangerous statement. Maybe it makes sense to use SCYLLA_ASSERT in some areas where failure is completely unrecoverable and maybe (?) it applies here, but this statement sounds very general, and I think the opposite is true in general - in general you should avoid SCYLLA_ASSERT like the plague - including for "things we know can't happen".
If you don't understand why I'm saying this, please refer to the issue I opened on this - https://github.com/scylladb/scylladb/issues/7871 - where I explained what I mean (this was already five years ago).